src/jarabe/journal/listview.py

```diff
@@ -747,7 +747,11 @@ def __icon_clicked_cb(self, cell, path):
 
     def __cell_title_edited_cb(self, cell, path, new_text):
         iterator = self._model.get_iter(path)
-        self._model[iterator][ListModel.COLUMN_TITLE] = new_text
+        if new_text.isspace() or new_text == '':
+            return
+        else:
+            self._model[iterator][ListModel.COLUMN_TITLE] = new_text
+
         self.emit('title-edit-finished')
```

Comment:
You need to make sure that the "title-edit-finished" is always emitted when the textbox is dismissed.  Currently, if the text is set to "" (emtpy) the function returns and the signal is not emitted.

The signal deals with auto-focusing the search bar when you type.  To see that feature in action, open the journal and start typing - it will type into the search bar.  However, this function is broken if the title-edit-finished is not emitted, see the steps to reproduce that I provided in the previous comment.